prov/psm2/src/psmx2_init.c

```diff
@@ -97,7 +97,7 @@ static int psmx2_init_lib(int default_multi_ep)
 		goto out;
 
 	if (default_multi_ep && !getenv("PSM2_MULTI_EP"))
-		putenv("PSM2_MULTI_EP=1");
+		putenv(strdup("PSM2_MULTI_EP=1"));
```

Comment:
What do you think about using the another POSIX-compliant variant [`setenv`](https://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/setenv.c) that makes a copy of the provided string and doesn't use the parameter directly in contrast to [`putenv`](https://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/putenv.c)? The remaining part of the psm/psm2 providers uses `setenv`

Reply:
Good catch. Thanks.